src/jdk.net/macosx/native/libextnet/MacOSXSocketOptions.c

```diff
@@ -198,42 +187,37 @@ JNIEXPORT jboolean JNICALL Java_jdk_net_MacOSXSocketOptions_ipDontFragmentSuppor
 (JNIEnv *env, jobject unused) {
     jint rv, fd, value;
     fd = socket(AF_INET, SOCK_DGRAM, 0);
-    if (fd == -1)
-        return JNI_FALSE;
-    value = 1;
-    rv = setsockopt(fd, IPPROTO_IP, IP_DONTFRAG, &value, sizeof(value));
-    close(fd);
-    if (rv == -1) {
-        return JNI_FALSE;
+    if (fd != -1) {
+        value = 1;
+        rv = setsockopt(fd, IPPROTO_IP, IP_DONTFRAG, &value, sizeof(value));
+        close(fd);
+        if (rv == -1) {
+            return JNI_FALSE;
+        }
     }
     fd = socket(AF_INET6, SOCK_DGRAM, 0);
-    if (fd == -1)
-        return JNI_FALSE;
-    value = 1;
-    rv = setsockopt(fd, IPPROTO_IPV6, IPV6_DONTFRAG, &value, sizeof(value));
-    close(fd);
-    if (rv == -1) {
-        return JNI_FALSE;
+    if (fd != -1) {
+        value = 1;
+        rv = setsockopt(fd, IPPROTO_IPV6, IPV6_DONTFRAG, &value, sizeof(value));
+        close(fd);
+        if (rv == -1) {
+            return JNI_FALSE;
+        }
     }
     return JNI_TRUE;
```

Comment:
Hello Daniel, if I'm reading this diff correctly, then we have inverted the check on `fd`. So I suspect this final `return` now needs to be changed to `return JNI_FALSE` to account for `fd` being `-1`?

Reply:
Hi Jaikiran, the point of this change was to better support systems where either IPv4 or IPv6 was disabled. I'm not sure if that can even happen on Mac. 
The change here is that if we can't create one socket, we don't immediately return false, but instead return the result for the other socket. For example, if creating IPv4 socket fails, we only check if DF is supported on IPv6 socket, and return that as a result. If both `socket` calls fail, we don't know if the DF flag is supported, and return true (before the patch we would return false in that case). I can modify the patch to return false in that case, or roll back this change entirely if you think it makes no sense.

Reply:
I think this change makes sense so long as we are sure that if IPv4 or IPv6 is disabled then the right call happens for the other option.

Reply:
The method will return true if the process is at the limit of open file descriptors. Before this patch the code returned false in that case.
As soon as Mac OS 10 (Catalina) goes out of support, we can remove this code entirely - AFAICT all newer versions support DF.